testsuite/MDAnalysisTests/core/test_requires.py

```diff
@@ -43,3 +43,20 @@ def mass_multiplier(ag1, ag2, scalar):
         result = mass_multiplier(u.atoms[:10], u.atoms[20:30], 4.0)
 
         assert_(isinstance(result, np.ndarray))
+
+    def test_failure_errormessage(self):
+        # failures should list all required attributes not
+        # just the first one
+        @requires('cats', 'dogs', 'frogs')
```

Comment:
:+1: It is great!

It is odd that the decorator needs 3 nested functions, but it happens with decorator sometimes.